text/0000-rust-has-provenance.md

```diff
@@ -0,0 +1,269 @@
+- Feature Name: rust_has_provenance
+- Start Date: 2023-11-22
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Pointers (this includes values of reference type) in Rust have **two** components.
+* The pointer's "address" says where in memory the pointer is currently pointing.
+* The pointer's "provenance" says where in memory the pointer is allowed to access when.
+
+(This is disregarding any "metadata" that may come with wide pointers, it only talks about thin pointers / the data part of a wide pointer.)
+
+Accessing memory using a pointer with incorrect provenance causes undefined behavior (UB), regardless of the address value of the pointer.
```

Comment:
But the validity of memory accesses do depend on the address. Provenance, if I understand correctly, describes a set of addresses which are valid to access.

Incorrectness of the provenance for an access depends on the address...

Not sure how to phrase this better though

Reply:
I have updated the wording; can you take a look?

Reply:
The new wording sound good to me 💜